### ISSUE TITLE
Two accounts with the same username can't be logged on to the same server.

### ISSUE THREAD

**Issue by fall** [SpigotMC/BungeeCord#3002]
There are a handful of accounts with the same username for example:
Chevy:4705f423623741fca7daa8bf76393de6
Chevy:cc5f347e4ec74a6da5e28f43a408e658
and when logging on to a BungeeCord server on both of the accounts, one of them will get disconnected with the error "You are already connected to this proxy".

This check should be done via UUID, not username.

**Comment by Janmm14** [comment]
Name duplications are not supported by vanilla minecraft servers and all existing name duplications are result of bugs on mojangs end.

Mojang does not support multiple accounts having the same username at the same time.

Likely one of those accounts still exists for the API but is banned.

Additionally such a change would hurt plugins as well as bungeecords internal commands. Bungeecord and the ecosystem around it is built with unique online player usernames in mind.

Bungeecord works as intended and this check stops plugins from breaking when mojang doesn't get their username uniqueness requirement (which they still have in place) enforced.

**Comment by Sagishii** [comment]
Recently, there was a bug in which it was possible to create duplicated Minecraft usernames (see: https://bugs.mojang.com/projects/WEB/issues/WEB-3566?filter=allopenissues&orderby=created+DESC%2C+priority+DESC%2C+updated+DESC)

When Mojang realized that they had created a dupe exploit, they had only reverted the newly made dupes and kept the old duped accounts the same without touching them at all. From this, I'm inferring that Mojang still "supports" old duped accounts as it was a mistake on their end which resulted in their creation years ago.

As most plugins are UUID-based (as they should be), I don't see how this is breaking anything. Also, both of the "Chevy" accounts mentioned above can be logged into and both have the complete capabilities of any regular Minecraft account (and are not just visible on the API)

**Comment by Janmm14** [comment]
Many if not all plugins rely on usernames to be unique at any given point.
Commands usually allow specifying a player by his name.

Additionally, bungeecord itself relies on names to be case-insensitive unique in multiple places, starting at:
https://github.com/SpigotMC/BungeeCord/blob/master/proxy/src/main/java/net/md_5/bungee/BungeeCord.java#L132

This is the main connection map bungeecord uses.

I don't see why anyone should revisit the whole bungeecord code to account fora quirk only a fraction of players will ever encounter.

Also I'm 99% sure the vanilla server doesn't allow that as well.

**Comment by md-5** [comment]
> Also I'm 99% sure the vanilla server doesn't allow that as well.

This

**Comment by fall** [comment]
Nope, they can both be logged onto vanilla servers, 100%. I've tested it myself.

**Comment by Sagishii** [comment]
I too have tried and it works 100%. Also, I don't think not accounting for dupe accounts is the solution here.

**Comment by md-5** [comment]
>I too have tried and it works 100%.

How do you kick one of them?

**Comment by Sagishii** [comment]
You can kick via UUID if the server is using Spigot, but not in vanilla.

**Comment by md-5** [comment]
Spigot doesn't change vanilla commands

**Comment by Sagishii** [comment]
> Spigot doesn't change vanilla commands

oh in that case sorry lol i didn't realize i had some plugin which changed the kick command in my test spigot server

**Comment by theodorefiji** [comment]
Name duplications are in fact supported by vanilla servers, in the sense that both can join a vanilla server at the same time.
I feel that Bungeecord should follow this too and allow duplicated accounts to join the same server at the same time, in order to keep consistency.

**Comment by Janmm14** [comment]
I think it should not happen because bungeecord internally and plugins are not created to support this.

**Comment by reegs417** [comment]
As a server owner myself, I feel that they should be able to join together as it gets really annoying for my dev team to have to change the code just so they can join together.

**Comment by Janmm14** [comment]
> As a server owner myself, I feel that they should be able to join together as it gets really annoying for my dev team to have to change the code just so they can join together.

Did you heard about this feature called "pull requests" where others can suggest code changes

**Comment by elonmusksama** [comment]
As was mentioned before, they should have compatibility for dupe accounts are they are accepted by Mojang (mojang knowingly does not revert them) and they are perfectly usable on Vanilla servers and should server owners be given a choice in config or whatever it should be a default yes instead of just defaulting to no imo

**Comment by mbkchris** [comment]
I feel that it need to be consistent in that since vanilla servers support it, bungeecord should too

**Comment by debby-ryan** [comment]
I completely agree with this, although a small percentage of accounts have their dupe counterparts it would be nice to see consistency in spigot with Vanilla. Vanilla servers support dupe accounts and although it seems like a hassle, it is a change that needs to come.

**Comment by md-5** [comment]
>it is a change that needs to come.

Be the change you wish to see in the world

**Comment by Sagishii** [comment]
> > it is a change that needs to come.
> 
> Be the change you wish to see in the world

you think this issue would be able to be fixed? @md-5

**Comment by Janmm14** [comment]
> > > it is a change that needs to come.
> > 
> > 
> > Be the change you wish to see in the world
> 
> you think this issue would be able to be fixed? @md-5

He was referring to the ability of Pull Requests

**Comment by jaylawl** [comment]
i do not recommend adding this ability and vanilla servers shouldn't allow duplicate players either with the current state of things.

lets assume you have 2 players with the same name on your server and one of them is a troublemaker that is harming your server in some way. be it just griefing, extremely malicious language or worse things... then how do you kick or ban them? do you kick/ban both players with the same name? not to mention that you will have trouble doing so via commands, because you can only target ONE of the SAME usernames via command (after all, commands are 99 % player name based and not UUID). And even if it were the latter... good luck getting the player's UUID in that moment.

i have my little admin tools plugin set up to not allow players to connect if there already is a player with the same name on the server. same goes for the very very few players that have unsafe characters in their usernames (like spaces, also an issue for command targeting)

just my two cents

**Comment by Sagishii** [comment]
> i do not recommend adding this ability and vanilla servers shouldn't allow duplicate players either with the current state of things.
> 
> 
> 
> lets assume you have 2 players with the same name on your server and one of them is a troublemaker that is harming your server in some way. be it just griefing, extremely malicious language or worse things... then how do you kick or ban them? do you kick/ban both players with the same name? not to mention that you will have trouble doing so via commands, because you can only target ONE of the SAME usernames via command (after all, commands are 99 % player name based and not UUID). And even if it were the latter... good luck getting the player's UUID in that moment.
> 
> 
> 
> i have my little admin tools plugin set up to not allow players to connect if there already is a player with the same name on the server. same goes for the very very few players that have unsafe characters in their usernames (like spaces, also an issue for command targeting)
> 
> 
> 
> just my two cents
> 
> 
> 
> /edit:
> 
> that all being said, optional functionality never hurts. in that sense: doesn't hurt to make this ability a togglable function in the config

If one of the dupe accounts were to be griefing on the server and the other one happened to be online at the same time, you could just run the kick command twice on the same username to temporarily kick/ban both of them and then fetch the UUID and then ban the right one with UUID. Also, dupe account's existance has been reported to Mojang several hundreds if not thousands of times and Mojang repeatedly does not do anything about it and allows them both to join Vanilla servers (probably like I mentioned before, it was mojang's fault that the duped accounts even existed in the first place)

**Comment by jaylawl** [comment]
> 
> 
> > i do not recommend adding this ability and vanilla servers shouldn't allow duplicate players either with the current state of

Sure, of course you could do that. It comes down to preference. Like i wouldn't want to put in that work (and additionally have to instruct server staff for such nieche cases). As said, i prefer the way of blocking identical usernames - and that puts a tiny bit of pressure back onto Mojang to fix it / not let this happen in the first place

**Comment by UnknownCreationsDev** [comment]
I'm currently working on this :)

**Comment by LeongYuZhi** [comment]
When can we expect this? This change needs to come. I'm a server owner and two of my players have "dupe accounts" like these and we would like them to be able to play on the server together as well.

**Comment by UnknownCreationsDev** [comment]
> When can we expect this? This change needs to come. I'm a server owner and two of my players have "dupe accounts" like these and we would like them to be able to play on the server together as well.

https://github.com/SpigotMC/BungeeCord/pull/3009
we're just waiting for the PR to be accepted

**Comment by Sagishii** [comment]
> > When can we expect this? This change needs to come. I'm a server owner and two of my players have "dupe accounts" like these and we would like them to be able to play on the server together as well.
> 
> #3009
> we're just waiting for the PR to be accepted

Yup

**Comment by beamwastaken** [comment]
How long does this need to get accepted? Seems quite urgent since it's completely stopping certain players from both joining the same server which could be a potential issue in larger servers and I don't think much other work is needed in the PR tbh.
